test/device/test_device_configuration.py

```diff
@@ -161,6 +161,84 @@ def test_can_clone_device(moler_config, device_factory):
+    moler_config.load_config(config=conn_config, config_type='yaml')
+
+    device_org_name = 'UNIX_LOCAL'
+    device_org = device_factory.get_device(name=device_org_name)
+    assert device_org is not None
+    device_org.remove()
+    with pytest.raises(KeyError):
+        device_factory.get_device(name=device_org_name)
+    clear_all_cfg()
+    moler_config.load_config(config=conn_config, config_type='yaml')
+    device_2 = device_factory.get_device(name=device_org_name)
+    assert device_2 != device_org
+
+
+def test_clone_device_from_cloned_device_device(moler_config, device_factory):
+    conn_config = os.path.join(os.path.dirname(__file__), os.pardir, "resources", "device_config.yml")
+    moler_config.load_config(config=conn_config, config_type='yaml')
+
+    device_org_name = 'UNIX_LOCAL'
+    device_cloned_name = 'UNIX_LOCAL_CLONED'
+    device_recloned_name = 'UNIX_LOCAL_RECLONED'
+    device_org = device_factory.get_device(name=device_org_name)
+    assert device_org is not None
+    device_cloned = device_factory.get_cloned_device(source_device=device_org, new_name=device_cloned_name)
+    device_recloned = device_factory.get_cloned_device(source_device=device_cloned, new_name=device_recloned_name)
+    assert device_cloned != device_recloned
+    assert device_cloned != device_org
+    assert device_recloned != device_org
+    device_cloned.remove()
+    device_recloned.remove()
+    with pytest.raises(KeyError):
+        device_factory.get_device(name=device_cloned_name)
+    with pytest.raises(KeyError):
+        device_factory.get_device(name=device_recloned_name)
+    with pytest.raises(KeyError):  # Cannot clone forgotten device.
+        device_factory.get_cloned_device(source_device=device_cloned_name, new_name=device_recloned_name)
+    with pytest.raises(KeyError):  # Cannot clone even passed reference to forgotten device.
+        device_factory.get_cloned_device(source_device=device_cloned, new_name=device_recloned_name)
+
+
+def test_clone_and_forget_device(moler_config, device_factory):
+    conn_config = os.path.join(os.path.dirname(__file__), os.pardir, "resources", "device_config.yml")
+    moler_config.load_config(config=conn_config, config_type='yaml')
+
+    device_org_name = 'UNIX_LOCAL'
+    device_cloned_name = 'CLONED_UNIX_LOCAL_TO_FORGET'
+    device_org = device_factory.get_device(name=device_org_name)
+    assert device_org is not None
+    with pytest.raises(KeyError):
+        device_factory.get_device(name=device_cloned_name)
+    device_cloned = device_factory.get_cloned_device(source_device=device_org, new_name=device_cloned_name)
+    assert device_cloned is not None
+    device_cloned.goto_state('UNIX_LOCAL')
+    cmd_ping = device_cloned.get_cmd(cmd_name="ping", cmd_params={"destination": 'localhost', "options": "-w 3"})
+    cmd_ping.start()
+    device_cloned.remove()
+    with pytest.raises(WrongUsage):
```

Comment:
Shouldn't we test also what is stated inside `WrongUsage`?
```
with pytest.raises(WrongUsage) as err:
    ...
assert "XXX" in str(err.value)
```

Reply:
Done